xyz-psql-connector/src/main/java/com/here/xyz/psql/PSQLXyzConnector.java

```diff
@@ -129,38 +129,73 @@ protected XyzResponse processGetFeaturesByBBoxEvent(GetFeaturesByBBoxEvent event
 
       final BBox bbox = event.getBbox();
 
-      if(event.getTweakType() != null)
-      { Map<String, Object> tweakParams = event.getTweakParams();
+      boolean bTweaks = ( event.getTweakType() != null ),
+              bOptViz = "viz".equals( event.getOptimizationMode() );
 
-        switch (event.getTweakType().toLowerCase()) {
+      if( !bOptViz && event.getSelection() != null && "*".equals( event.getSelection().get(0) ) ) //correction for raw usecase
+       event.setSelection(null);
+
+      if( bTweaks || bOptViz )
+      { String tweakType;
+        Map<String, Object> tweakParams;
+        boolean bVizSamplingOff = false;
+
+        if( bTweaks )
+        { tweakType   = event.getTweakType().toLowerCase();
+          tweakParams = event.getTweakParams();
+        }
+        else
+        { tweakType   = TweaksSQL.ENSURE;
+          event.setTweakType( TweaksSQL.ENSURE );
+          tweakParams = new HashMap<String, Object>();
+          switch( event.getVizSampling().toLowerCase() )
+          { case "high" : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 10 ) ); break;
+            case "low"  : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 70 ) ); break;    
+            case "off"  : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 100 )); 
+                          bVizSamplingOff = true;
+                          break;    
+            case "med"  : 
+            default     : tweakParams.put(TweaksSQL.ENSURE_SAMPLINGTHRESHOLD, new Integer( 20 ) ); break;    
+          }
+        }
+
+        int distStrength = 0;
+
+        switch (tweakType) {
 
           case TweaksSQL.ENSURE: {
             int rCount = executeQueryWithRetry(SQLQueryBuilder.buildEstimateSamplingStrengthQuery(event, bbox )).getFeatures().get(0).get("rcount");
 
             boolean bDefaultSelectionHandling = (tweakParams.get(TweaksSQL.ENSURE_DEFAULT_SELECTION) == Boolean.TRUE );
 
-            if( event.getSelection() == null && !bDefaultSelectionHandling )
+            if( event.getSelection() != null && "*".equals( event.getSelection().get(0) ) )
```

Comment:
if suggestion on line 135 is taken, these lines 171&172 and  should be removed and line 173 adjusted accordingly.

Reply:
s.o. evaluation of new introduced "*" value.